apps/examples/testcase/le_tc/filesystem/fs_main.c

```diff
@@ -2445,6 +2445,121 @@ static void tc_libc_stdio_stdsostream(void)
 }
 
 /**
+* @testcase         tc_libc_stdio_tempnam_tc
+* @brief            Returns a pointer to a string that is a valid filename
+* @scenario         The tempnam() function returns a pointer to a unique temporary filename, or NULL if a unique name cannot be generated.
+* @apicovered       tempnam(), mkstemp(), mktemp()
+* @precondition     NA
+* @postcondition    NA
+*/
+
+static void tc_libc_stdio_tempnam_tc(void)
+{
+	const char filename[] = "tempnam_tc_test";
+	char *ret1 = "";
+	char *ret2 = "";
+	int ret_check = 0;
+	FILE *fp;
+
+	/* TC 1 with valid prefix */
+
+	ret_check = mount(MOUNT_DEV_DIR, CONFIG_LIBC_TMPDIR, "smartfs", 0, NULL);
+	TC_ASSERT_EQ("mount", ret_check, OK);
+
+	ret1 = tempnam(CONFIG_LIBC_TMPDIR, filename);
+	TC_ASSERT_NEQ_CLEANUP("tempnam", ret1, NULL, umount(CONFIG_LIBC_TMPDIR));
+
+	fp = fopen(ret1, "w");
+	TC_ASSERT_NEQ_CLEANUP("fopen", fp, NULL, goto errout1);
```

Comment:
This case is same with what I said before.
If fp == NULL, it means that file is not valid and we don't need to unlink.

Reply:
@jeongarmy 
i think once tempnam is successfull, we will need to unlink the file created.
let me know your opinion